samples/detect.v2beta1.js

```diff
@@ -0,0 +1,728 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const projectId = process.env.GCLOUD_PROJECT;
+const sessionId = require('uuid/v1')();
+const common = require('@google-cloud/common');
+const structjson = require('./structjson.js');
+
+function createKnowledgeBase(projectId, displayName) {
+  // [START dialogflow_create_knowledge_base]
+  // Imports the Google Cloud DialogFlow library
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow client.
+  var client = new dialogflow.KnowledgeBasesClient();
+
+  var formattedParent = client.projectPath(projectId);
+  var knowledgeBase = {
+    displayName: displayName,
+  };
+  var request = {
+    parent: formattedParent,
+    knowledgeBase: knowledgeBase,
+  };
+
+  client
+    .createKnowledgeBase(request)
+    .then(responses => {
+      const result = responses[0];
+      console.log(`Name: ${result.name}`);
+      console.log(`displayName: ${result.displayName}`);
+      return result.displayName;
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END dialogflow_create_knowledge_base]
+}
+
+function getKnowledgeBase(projectId, knowledgeBaseFullName) {
+  // [START dialogflow_get_knowledge_base]
+  // Imports the Google Cloud DialogFlow library
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow client.
+  var client = new dialogflow.KnowledgeBasesClient();
+
+  const fullName = knowledgeBaseFullName;
+
+  client
+    .getKnowledgeBase({
+      name: fullName,
+    })
+    .then(responses => {
+      const result = responses[0];
+      console.log(`Name: ${result.name}`);
+      console.log(`displayName: ${result.displayName}`);
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END dialogflow_get_knowledge_base]
+}
+
+function listKnowledgeBases(projectId) {
+  // [START dialogflow_list_knowledge_base]
+  // Imports the Google Cloud DialogFlow library
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow KnowledgeBasesClient.
+  var client = new dialogflow.KnowledgeBasesClient({
+    projectPath: projectId,
+  });
+
+  // Iterate over all elements.
+  var formattedParent = client.projectPath(projectId);
+
+  client
+    .listKnowledgeBases({parent: formattedParent})
+    .then(responses => {
+      var resources = responses[0];
+      for (let i = 0; i < resources.length; i += 1) {
```

Comment:
Is it possible to use forEach here?

Reply:
added!  Thanks I swipped that from docs- this comment might still appear as TODO comments will change line numbers and i can't close out comments-